src/test/xmeta/test_symbol_table.cpp

```diff
@@ -256,27 +263,81 @@ TEST_CASE("Struct test")
 
 TEST_CASE("Struct circular test")
 {
-    std::istringstream struct_test_idl{ R"(
-        namespace N
-        {
-            struct S1
+    {
+        std::istringstream struct_test_idl{ R"(
+            namespace N
             {
-                S2 field_1;
+                struct S0
+                {
+                    S1 field_1;
+                }
+                struct S1
+                {
+                    S2 field_1;
+                }
+                struct S2
+                {
+                    S3 field_1;
+                }
+                struct S3
+                {
+                    S1 field_1;       
+                }
             }
-            struct S2
+        )" };
+
+        xmeta_idl_reader reader{ "" };
+        reader.read(struct_test_idl);
+        REQUIRE(reader.get_num_syntax_errors() == 0);
+        REQUIRE(reader.get_num_semantic_errors() == 4);
```

Comment:
Consider just testing for >0 here so that it's not fragile with respect to how the error is expressed.